### PR TITLE
Build all four ROMs in the dump

### PR DESCRIPTION
**Pull request by PikalaxALT**
Upon merge, the following ROMs can be built without error:
- Pokemon Gold Spaceworld (`3c407114de28d17b7113a2c0cee9a37c`)
- Pokemon Gold Debug Spaceworld (`2eadbed83b775c097ff79e5128d1184f`)
- Pokemon Silver Spaceworld (`c52a677c35f15320d5b495e14809f00d`)
- Pokemon Silver Debug Spaceworld (`fa65d3759bb17c489de171a598ba4913`)

At present, an issue remains with the global checksum correction for the `-correctheader.gb` ROMs when using rgbfix 0.3.7.

### CODE REVIEW

**Comment by obskyr** [comment]
Oh, what's the rgbfix issue about?

**Comment by ISSOtm** [comment]
The Debug Silver ROM with fixed header originally uploaded to TCRF had its checksums not fixed by mistake. The properly fixed version should have hash `ee3dcac13383404bd4178b4ca34e7e73`.

File: data/trainers/parties.asm [hunk 1]
```diff
@@ -3,10 +3,10 @@ INCLUDE "constants.asm"
...
 ; - db "NAME@", TRAINERTYPE_* constant
 ; - 1 to 6 Pokémon:
-;    * for TRAINERTYPE_NORMAL:     db level, species
-;    * for TRAINERTYPE_ITEM:       db level, species, item
-;    * for TRAINERTYPE_MOVES:      db level, species, 4 moves
-;    * for TRAINERTYPE_ITEM_MOVES: db level, species, item, 4 moves
+;	* for TRAINERTYPE_NORMAL:	 db level, species
+;	* for TRAINERTYPE_ITEM:	   db level, species, item
+;	* for TRAINERTYPE_MOVES:	  db level, species, 4 moves
+;	* for TRAINERTYPE_ITEM_MOVES: db level, species, item, 4 moves
```
Comment (dannye) [inline data/trainers/parties.asm]: This is part of the reason we do spaces for alignment

File: Makefile [hunk 2]
```diff
@@ -52,20 +55,20 @@ mostlyclean:
...
-
-$(BUILD)/shim.asm: tools/make_shim.py shim.sym | $$(dir $$@)
-	$(PYTHON) tools/make_shim.py -w -- $(filter-out $<, $^) > $@
+	$(RGBFIX) -f hg -m 0x10 $@
 
 .PHONY: coverage
-coverage: $(ROMS)
-	$(PYTHON) tools/disasm_coverage.py -m $(ROMS:.gb=.map) -b 0x40
+coverage: $(ROM)
+	$(PYTHON3) tools/disasm_coverage.py -m $(ROM:.gb=.map) -b 0x40
```
Comment (dannye) [inline Makefile]: You've missed some discussion be we are trying to stick with version agnostic python

File: data/trainers/parties.asm (hunk 1 above)
Comment (PikalaxALT) [reply data/trainers/parties.asm]: Oh, okay. I was trained to do tabs for asm.

File: Makefile (hunk 2 above)
Comment (PikalaxALT) [reply Makefile]: Clearly not doing a good job, the `encoding` keyword to `open` is a Python 3 exclusive.

**Comment by mid-kid** [comment]
Use `sort_symfile.sh` from pokecrystal instead of `sort_sym.py`.

**Comment by mid-kid** [comment]
Consider a makefile like this: https://ghostbin.com/paste/7jdsy
i.e.: GFX aren't version-dependent, only .o files are built in a version-dependant manner, by prefixing them with the correct build directory. Doesn't require recursive make.

**Comment by PikalaxALT** [comment]
After heated discussion, we have decided to forego multiple ROM building for the time being, but will retain the disassembled version differences present in this PR.

**Comment by ISSOtm** [comment]
This doesn't build the "fixed header" ROM correctly due to the RGBFIX 0.3.7 bug, but otherwise this worked OK on my machine. Merging.

**Pull request merged**